src/routes/measurement-threshold/find.ts

```diff
@@ -0,0 +1,111 @@
+import Joi from 'joi';
+import { Edge } from '../../models/edge';
+import { Pin } from '../../models/pin';
+import { ControllerGeneratorOptionsWithClientAndSupplier } from '../../comms/controller';
+
+import { PinGroup } from '../../models/pin-group';
+
+import { TableQuery, EffectiveTableQuery, tableQuerySchemaGenerator } from '../../comms/table-controller';
+import { Quantity } from '../../models/quantity';
+import { stringBeforeV7ElseStringOrTranslationSchema } from '../../models/string-or-translations';
+import { schema as siNumberSchema, SiNumber } from '../../models/si-number';
+
+interface Query extends TableQuery {
+  quantityHashId?: string | null;
+  pinGroupHashId?: string | null;
+  edgeHashId?: string | null;
+  pinHashId?: string | null;
+  type?: 'quantity' | 'port';
+}
+
+type Request = {
+  query?: Query;
+} | undefined;
+
+interface EffectiveQuery extends EffectiveTableQuery {
+    quantityHashId?: string | null;
+    pinGroupHashId?: string | null;
+    edgeHashId?: string | null;
+    pinHashId?: string | null;
+    type: 'quantity' | 'port';
+}
+
+interface EffectiveRequest {
+  query: EffectiveQuery;
+}
+
+interface Threshold {
+  criticallyLow: SiNumber | null;
+  low: SiNumber | null;
+  high: SiNumber | null;
+  criticallyHigh: SiNumber | null;
+}
+
+interface ResponseRow {
+  quantity: Pick<Quantity, 'name' | 'hashId'>;
+  location: {
+    pinGroup: Pick<PinGroup, 'name' | 'hashId'>;
+    pin: Pick<Pin, 'name' | 'hashId'>;
+    edge: Pick<Edge, 'name' | 'hashId'> | null;
+    } | null;
+  threshold: Threshold;
+}
+
+interface Response {
+  nextPageOffset: string | null;
+  rows: ResponseRow[];
+}
+
+const controllerGeneratorOptions: ControllerGeneratorOptionsWithClientAndSupplier = {
+  method: 'get',
+  path: '/',
+  query: tableQuerySchemaGenerator(Joi.string().valid('quantityName', 'pinGroupName', 'hashId').default('quantityName')).keys({
+    quantityHashId: Joi.string().allow(null).default(null),
+    pinGroupHashId: Joi.string().allow(null).default(null),
+    edgeHashId: Joi.string().allow(null).default(null),
+    pinHashId: Joi.string().allow(null).default(null),
+    type: Joi.string().valid('quantity', 'port'),
+  }),
+  // rights to be updated to 'Threshold' which will be added soon
```

Comment:
please add the ticket number for the todo

Reply:
Added the rights, no longer needed.